src/main/java/spoon/support/visitor/ClassTypingContext.java

```diff
@@ -643,7 +646,7 @@ private boolean isSubTypeArg(CtTypeReference<?> subArg, CtTypeReference<?> super
 		if (typeRef instanceof CtTypeParameterReference) {
 			CtTypeParameterReference typeParamRef = (CtTypeParameterReference) typeRef;
 			CtTypeParameter typeParam = typeParamRef.getDeclaration();
-			if (typeParam.getTypeParameterDeclarer() instanceof CtExecutable) {
+			if (typeParam != null && typeParam.getTypeParameterDeclarer() instanceof CtExecutable) {
 				//the parameter is declared in scope of Method or Constructor
```

Comment:
It was happening in `LambdaTest.testCompileLambdaGeneratedBySpoon` and `MainTest.testTest`. Apparently the problem is it tries to resolve `R` in `public abstract R apply(T arg0);` but it never reach the interface `Function` where it's defined.